central/reports/common/query_builder.go

```diff
@@ -40,9 +42,17 @@ func NewVulnReportQueryBuilder(collection *storage.ResourceCollection, vulnFilte
 }
 
 // BuildQuery builds scope and cve filtering queries for vuln reporting
-func (q *queryBuilder) BuildQuery(ctx context.Context) (*ReportQuery, error) {
-	// TODO ROX-18773 : Add SAC query filter on deploymentsQuery to scope the results by report requester.
+func (q *queryBuilder) BuildQuery(ctx context.Context, clusters []*storage.Cluster,
+	namespaces []*storage.NamespaceMetadata) (*ReportQuery, error) {
 	deploymentsQuery, err := q.collectionQueryResolver.ResolveCollectionQuery(ctx, q.collection)
+	if env.VulnReportingEnhancements.BooleanSetting() {
+		scopeQuery, err := q.buildAccessScopeQuery(clusters, namespaces)
+		if err != nil {
+			return nil, err
+		}
+		deploymentsQuery = search.ConjunctionQuery(deploymentsQuery, scopeQuery)
+	}
+
```

Comment:
We discussed on the call that v1 and v2 can co-exist if 2.0 is a tech preview. Therefore, we want if to **not execute** if the request is from v1. Is `buildAccessScopeQuery` backward compatible? If not, I suggest you create two copies of the file, one for v1 and v2.

Reply:
`buildAccessScopeQuery` has the following check
```
if accessScopeRules == nil {
		// Old(v1) report configurations would have nil access scope rules.
		// For backward compatibility, nil access scope would mean access to all clusters and namespaces.
		// To deny access to all clusters and namespaces, the accessScopeRules should be empty.
		return search.EmptyQuery(), nil
	}
``` 

This should take care of v1 report configs because accessScopeRules will be nil for them. It is backward compatible and we actually do not need the `if env.VulnReportingEnhancements.BooleanSetting()` check.